Text proposal for the Survey on the Impact of Allowing HTML Syntax in EPUB (#21)
w3c/pm-wg: survey/HTML-syntax-in-EPUB.md

---
title: Survey on the Impact of Allowing HTML Syntax in EPUB
date: 2025-05-08
---

> The Publishing Maintenance Working Group (PMWG) is exploring the possibility of allowing HTML as an alternative content document format in EPUB 3.4, alongside the existing XHTML format.

Reviewer: ```suggestion
The Publishing Maintenance Working Group (PMWG) is exploring the possibility of allowing the [HTML syntax](https://html.spec.whatwg.org/multipage/syntax.html#syntax) in the HTML content document format in EPUB 3.4, and not restricting it to the [XML Syntax](https://html.spec.whatwg.org/multipage/xhtml.html#the-xhtml-syntax) as it is in earlier versions of EPUB 3.
```

The reference to the "syntax" was mentioned on the call...

Reviewer 2: It might help to explain what we're talking about before jumping into the new formats we're allowing, and we should use xml and html syntaxes consistently otherwise we're just going to sow confusion. I'd suggest a simple intro like:

HTML(5) defines both XML and HTML syntaxes for authoring documents. EPUB 3, however, currently restricts authors to using only the XML syntax for content documents. The Publishing Maintenance Working Group (PMWG) is considering expanding EPUB content to allow both syntaxes to be used to author content documents in EPUB 3.4.

> This initiative aims to address the diminishing use and maintenance of XHTML outside the publishing industry while leveraging HTML's status as a living standard. 

Reviewer: ```suggestion
This initiative aims to address the diminishing use and maintenance of the XML syntax outside the publishing industry while leveraging the future evolution of HTML. 
```

The "Living standard" terminology is not necessarily known outside the standard world, and may be frightening for outsiders...

> The proposal does not fundamentally alter EPUB's reliance on HTML5 but removes the restriction to XML syntax. This change could simplify workflows, reduce barriers to entry, and potentially broaden adoption across the ebook value chain—from content creation to consumption.

Reviewer: ```suggestion
The proposal does not alter EPUB's reliance on HTML; it merely removes the restriction to the XML syntax. This change could simplify workflows, reduce barriers to entry, and potentially broaden adoption across the ebook value chain—from content creation to consumption. It also makes it possible to use tools, run-time libraries, programming methodologies, etc., developed by the Web community and which are not necessarily prepared for the XML syntax of HTML.
```

I wonder, though, whether the first sentence is necessary. Doesn't it repeat what has been said in the previous paragraph? (Although re-emphasizing might make sense.)

> PMWG is seeking your valuable feedback. What can you see as potential benefits? Do you anticipate any technical challenges? What would you consider a reasonable implementation timeline?  Your insights will help shape the future of EPUB standards. 

Reviewer 3: We should probably avoid posing these as questions to avoid the temptation of a responding to the mail itself. Maybe:

PMWG is seeking your valuable feedback with regards to potential benefits, technical challenges, and implementation timelines.  Your insights will help shape the future of EPUB standards.

> Your responses will be kept anonymous but will be made public to ensure transparency. Providing some context about your tools and workflows will help us better understand and address your feedback.

Reviewer 2: We should ask for permission to publish comments. This isn't directly the work of the PMWG so transparency should only require that we summarize the feedback that people don't want published so that we're not accused of suppressing feedback we don't like.

In other words, add a checkbox question like: "Do you consent to having your comments reproduced publicly? Note, if you answer no your feedback may still be summarized and reproduced for reporting purposes but no personally identifiable information will be included."

> Your responses will be kept anonymous but will be made public to ensure transparency. Providing some context about your tools and workflows will help us better understand and address your feedback.

Author: I understood that the w3c survey tool makes outputs publics.

> Your responses will be kept anonymous but will be made public to ensure transparency. Providing some context about your tools and workflows will help us better understand and address your feedback.

Reviewer: We do not know yet whether we can use the W3C tool (I have asked whether it is possible, but got no response yet).  Besides, it may be possible to make the responses and, primarily, the origin of the responses, confidential.

Let us forget about the tool, we can choose that later.

I think adding something like what @mattgarrish proposes:

> "Do you consent to having your comments reproduced publicly? Note, if you answer no your feedback may still be summarized and reproduced for reporting purposes but no personally identifiable information will be included."

is wise.

> **Question 1:** What challenges, if any, would your organization face in adding support for HTML syntax in your workflows? Please describe any technical, operational, or resource-related issues you foresee.

Reviewer 3: I am assuming this will move to whatever we tool we use to collect responses, and be replaced with a link. We should probably be clear that they should follow the link to respond and not just reply to the email.

> **Question 1:** What challenges, if any, would your organization face in adding support for HTML syntax in your workflows? Please describe any technical, operational, or resource-related issues you foresee.

Reviewer 3: Is the intent to add more questions? This only address challenges in benefits and timelines. Should those be added as new questions? E.g. "What benefits do you see from this change? What drawbacks do you see?" and "How much work would such a change be on your workflows and products?". And maybe and final "Other thoughts" section. Or do we want just a single open ended question?

---
title: Survey on the Impact of Allowing HTML Syntax in EPUB
date: 2025-05-08
---

The Publishing Maintenance Working Group (PMWG) plans to allow the [HTML syntax](https://html.spec.whatwg.org/multipage/syntax.html#syntax) for the HTML content document format in EPUB 3.4, instead of restricting it to the [XML Syntax](https://html.spec.whatwg.org/multipage/xhtml.html#the-xhtml-syntax) as it was done for earlier versions of EPUB 3.

> 

Reviewer 4: Isn't "plans to" too strong?

> The Publishing Maintenance Working Group (PMWG) is exploring the possibility of allowing the [HTML syntax](https://html.spec.whatwg.org/multipage/syntax.html#syntax) for the HTML content document format in EPUB 3.4, instead of restricting it to the [XML Syntax](https://html.spec.whatwg.org/multipage/xhtml.html#the-xhtml-syntax) as it is in earlier versions of EPUB 3.

Reviewer: ```suggestion
The Publishing Maintenance Working Group (PMWG) plans to allow the [HTML syntax](https://html.spec.whatwg.org/multipage/syntax.html#syntax) for the HTML content document format in EPUB 3.4, instead of restricting it to the [XML Syntax](https://html.spec.whatwg.org/multipage/xhtml.html#the-xhtml-syntax) as it was done for earlier versions of EPUB 3.
```

I think what we said on the call is that we do plan to do this, but we have to find the best way of doing it. Hence, the proposed change.

> This initiative aims to address the diminishing use and maintenance of the XML syntax outside the publishing industry while leveraging the future evolution of HTML. 

Reviewer 5: ```suggestion
XML use is diminishing outside of publishing. The syntax is no longer being maintained. No new features will be added to XML, and the syntax will not be updated even when related technologies change. By introducing HTML into the EPUB package, the format can continue to evolve. 
```

Reviewer 2: This sounds like a statement about XML generally. It would be good to stick to "XML syntax" and keep it tied to HTML so we don't ruffle any feathers.

Reviewer 5: That would make the statement more precise, thank you. 
How is this—

> Use of the XML syntax is diminishing outside of publishing. The syntax is no longer being maintained. No new features will be added, and the syntax will not be updated even when related technologies change. By introducing HTML into the EPUB package, ebooks can continue to evolve. 

note: I also removed “to the syntax” after “added” (to reduce repetition)
and changed “the format” to “ebooks" (for clarity)

Should we scare them a little bit here? It is implied that things might break as XML syntax gets out of sync with related tech. Should we say that more clearly?

Reviewer: > Should we scare them a little bit here? It is implied that things might break as XML syntax gets out of sync with related tech. Should we say that more clearly?

Not a bad idea but, just to soften the pill, make it clear that it is something that might happen on long term, not right now. Ie, making it somehow clearer that, with that move, we are building for the future here.

Otherwise, I am fine with the proposed text, thanks!

Reviewer 3: Nit: s/HTML/HTML syntax/ - we need to be clear that HTML itself  is already fully supported.

Reviewer 5: Does this explain the situation, outline the risk of doing nothing, and make it clear we are building for the future?

> Use of the XML syntax is diminishing outside of publishing. The syntax is no longer being maintained. No new features will be added, and the syntax will not be updated even when related technologies change. Eventually, relying on a deprecated syntax could cause errors and prevent innovation. By introducing HTML syntax into the EPUB package, ebooks can continue to evolve into the future.

Reviewer: Shouldn't it be "By introducing the HTML syntax..." (missing 'the')?

Otherwise I am fine with this.

Reviewer 5: ```suggestion
Use of the XML syntax is diminishing outside of publishing. The syntax is no longer being maintained. No new features will be added, and the syntax will not be updated even when related technologies change. Eventually, relying on a deprecated syntax could cause errors and prevent innovation. By introducing HTML syntax into the EPUB package, ebooks can continue to evolve into the future.
```

> The proposal does not change the document formats currently allowed in an EPUB. Since EPUB is already reliant on HTML, this proposal merely removes the restriction to the XML syntax.

Reviewer: ```suggestion
The change does not affect the document formats currently allowed in an EPUB publication; they will remain valid as they are today. This change merely removes the restriction to the XML syntax.
```

> PMWG is seeking your valuable feedback. What can you see as potential benefits? Do you anticipate any technical challenges? What would you consider a reasonable implementation timeline?  Your insights will help shape the future of EPUB standards. 

Reviewer: ```suggestion
PMWG is seeking your valuable feedback.  The purpose of the survey is to get the temperature of the community about this technical change, figure out the potential difficulties, and what the Working Group can do to make the transition easier.  Your insights will help shape the future of EPUB standards. 
```

(I copied the remark of @sueneu in the minutes.)

> PMWG is seeking your valuable feedback.  The purpose of the survey is to get the temperature of the community about this technical change, figure out the potential difficulties, and what the Working Group can do to make the transition easier.  Your insights will help shape the future of EPUB standards. 

Reviewer 5: ```suggestion
PMWG is seeking your valuable feedback. The purpose of this survey is to get the temperature of the community about the technical change. The Working Group hopes to discover potential obstacles so we can make the transition easier. Your insights will help shape the future of EPUB standards. 
```

Reviewer 3: I mean ... do we really *hope* to discover obstacles? I kind of hope we don't, though I expect we will. Maybe "The Working Group wants to identify any potential obstacles ..."

> PMWG is seeking your valuable feedback. The purpose of this survey is to get the temperature of the community about the technical change. The Working Group hopes to discover potential obstacles so we can make the transition easier. Your insights will help shape the future of EPUB standards. 

Reviewer 2: Instead of "hopes to discover potential obstacles" as discussed on the call today, could this be changed to "The Working Group will use your feedback to help identify any obstacles ..."?

```suggestion
PMWG is seeking your valuable feedback. The purpose of this survey is to get the temperature of the community about the technical change. The Working Group will use your feedback to help identify any potential obstacles so we can make the transition easier. Your insights will help shape the future of EPUB standards. 
```

Reviewer 2: Ya, I don't think there's a way to make a pull request that we'll see here off someone's local fork. Suggesting changes through the comments is the only way I know of as github handles committing it back to the fork. It looked like this had been forgotten which is why I suggested a change.

Author: I would have preferred to be able to branch the repo here, but I was not authorised.

Your responses will be kept anonymous but will be made public to ensure transparency. Providing some context about your tools and workflows will help us better understand and address your feedback.

**Question 1:** What is your position in the industry and how do your workflows involve EPUB?

**Question 2:** What potential benefits and challenges do you anticipate when incorporating HTML syntax into your processes? 

> **Question 3:** Are there any technical, operational, or resource related issues that need to be considered?

Reviewer: ```suggestion
**Question 3:** Are there any technical, operational, or resource related steps that the Working Group should consider to ease the transition (e.g., additional metadata, profiling the EPUB specification, introducing several EPUB versions, etc.)?
```

Author: Should we be that verbose and directive here? Simple questions are easier to understand.

Reviewer: Yes, that is correct, but I am a little bit afraid the answers will be all over the place. What about

> Are there steps that the WG should consider easing the transition (e.g., additional metadata, profiles of EPUB, etc.)?

> **Question 3:** Are there any technical, operational, or resource related issues that need to be considered?

Reviewer 3: Should we have a "Q4: Additional comments [optional]"?

Reviewer: Yep, that is a good idea. It shows openness...
